File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(_matrix) {
    this.rows = _matrix
      .split("\n")
      .map((r) => r.split(" ").map((n) => Number(n)));
    this.columns = this.rows[0].map((_, i) => this.rows.map((r) => r[i]));
  }
}

File: ruby/matrix/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end

class Matrix
  attr_reader :rows,
              :columns

  def initialize(text)
    @rows = text.lines.map { |line| line.split(' ').map(&:to_i) }
    puts "rows #{@rows}"
    @columns = rows.transpose
    puts "column #{@columns}"

  end

  def row(offset)
    rows[offset - 1]
  end

  def column(offset)
    columns[offset - 1]
  end
end

matrix = Matrix.new("1 2\n3 4")
